api/app.js

```diff
@@ -2,17 +2,39 @@ const express = require("express");
 const app = express();
 
 const { getCategories } = require("./controllers/categoriescontroller");
+const { getReview } = require("./controllers/reviewsIDcontroller");
 
 app.use(express.json());
 
 app.get("/api/categories", getCategories);
 
+app.get("/api/reviews/:review_id", getReview);
+
 //404 wrong path
 app.all("/api/*", (req, res, next) => {
   res.status(404).send({
     msg: "Page not found",
   });
 });
 
+//javascript
+
+app.use((err, req, res, next) => {
+  if (err.status) {
+    res.status(err.status).send({ msg: err.msg });
+  } else {
+    next(err);
+  }
+});
+
+// psql error
+
+app.use((err, req, res, next) => {
+  if (err.code.length === 5 && err.code) {
+    res.status(400).send({ msg: "bad request" });
+  } else {
+    next(err);
+  }
+});
 
```

Comment:
Smart handling of psql errors here with length, just be careful to handle specific errors as this could give you unexpected results for errors we don't want to handle this way.